proposals/LineStyle/README.md

```diff
@@ -1,76 +1,140 @@
+- patternScale. A float uniform. It is valid when the primitive binds a DashDotPattern. The default value is 1. You can lengthen or compress the line pattern by setting this property. For example, if patternScale is set to 2, the length of each dash and each gap will be enlarged by 2 times. This value will not impact on the line width.
+- screenspacePattern. A bool uniform. It is valid when the primitive binds a DashDotPattern. By default it is true, which means the dash-dot pattern will be based on screen unit. If we zoom in, the pattern on the line will change in the world space, so that the dash size and the dash gap size in the screen space will not change. If it is false, the pattern will be based on world unit. If we zoom in, the pattern on the line will not change in world space. The dash size and the dash gap size in the screen space will be larger. 
+
+![image of screenspacePattern](screenSpacePattern.png)
+
+### Extents of the DashDotLines
+Different from the other Curves, the extents of the DashDotLines is only the bound box of the control points. The width of the line will not be considered, because it is screen spaced, that it is implemented via the shader.
+
+### The DashDotPattern schema
+The DashDotPattern schema inherits from Typed. It saves the detail of a dash-dot pattern. A dash-dot pattern is an array of symbols. Each symbol is either a dash, which is a line, or a dot, which is a point. The DashDotPattern can be bound to one or more DashDotLines primitive.
+
+The DashDotPattern has the following new properties:
+- patternPeriod. A float uniform. It is the length of a pattern. By default it is zero. If there is no pattern, it should be zero.
+- pattern. An array of float2. It saves the dash-dot pattern. For each float2, the x value and the y value must be zero or positive. The x value saves the offset of the start of current symbol, from the end of the previous symbol. If the current symbol is the first symbol, the offset is from the start of the pattern to the start of current symbol. The y value saves the length of the current symbol. If it is zero, the current symbol is a dot. If it is larger than zero, the current symbol is a dash. As a result, the total sum of all the x value and y value will be the length from the start of the pattern to the end of the last symbol. This sum must be smaller than patternPeriod.
+        
+For example, assume the pattern is [(0, 10), (1, 4), (3, 0)]. It means the first symbol is a dash which is from 0 to 10. The second symbol is a dash which is from 11 to 15, and the third symbol is a dot which is at position 18. There are gaps between 10 and 11, and between 15 and 18. If the patternPeriod is 20, there is also a gap between 18 and 20.
+
+An API schema DashDotPatternAPI is provided to bind the DashDotPattern to a DashDotLines primitive.
```

Comment:
Specifying the Pattern as a separate prim that must be bound via relationship seems like overkill here, and imposes more complexity/cost on the scenegraph and Hydra efficiency than a simpler, API-only solution would.  UsdShadeMaterial must be bound because it consists of multiple prims, and because the binding mechanism itself needs to be very "rich" (i.e. complicated).

As an alternative, could we get rid of the `DashDotPattern` Typed schema, instead putting all of its properties into `DashDotPatternAPI`?  Then, the simple case is _very_ simple: all the information for a single DashDotLines is contained **on the prim**, very efficiently processed by Hydra.  In the mentioned case of wanting to "bind" the same Pattern onto multiple `DashDotLines` primitives, this seems like a CSS-inspired dream case for the use of class inheritance in USD.  

Now we can't build inheritance into the schema itself (composition doesn't consult the `UsdSchemaRegistry`), but we _can_ provide "strong guidance" in the spec (and helper functions in the schema classes) that `DashDotLines` primitives should minimally inherit from a "_class_DashDotPattern", and potentially also more specific/granular classes tailored to the asset/scene.  So, each of the `DashDotLine` primitives in the "Diagram1" asset might look something like:

```
def DashDotLines "TopArc" (
    prepend inherits = [</_class_Diagram1_DD_Pattern>, </_class_DashDotPattern>]
)
{
}
```

so then anyone referencing that asset could affect all of its `DashDotLines` (and no others) by putting opinions for `DashDotPatternAPI` in a spec for `</_class_Diagram1_DD_Pattern> `.

Reply:
Thank you. I will modify the proposal. And also for the text proposal, I will also merge the TextStyle into TextStyleAPI, and the same for ParagraphStyle and ColumnStyle.